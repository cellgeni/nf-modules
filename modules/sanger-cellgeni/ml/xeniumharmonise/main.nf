
process ML_XENIUMHARMONISE {
    tag "$meta"

    container '/nfs/cellgeni/singularity/images/palom_tiledb.sif'

    input:
    tuple val(meta), path(he_image), path(xenium_bundle)

    output:
    tuple val(meta), path(output_dir), emit: harmonised_output
    path "versions.yml"               , emit: versions

    script:

    output_dir = "${meta}_harmonised"
 
    """
    harmonise_xenium.py main\\
        --hematoxylin_eosin_image_uri ${he_image} \\
        --xenium_bundle_uri ${xenium_bundle} \\
        --harmonised_dataset_uri ${output_dir} \\

        
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mlxeniumharmonise: 1.0.0
    END_VERSIONS
    """

    stub:
    output_dir = "${meta}_harmonised"

    """
    touch ${output_dir}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mlxeniumharmonise: 1.0.0
    END_VERSIONS
    """
}
