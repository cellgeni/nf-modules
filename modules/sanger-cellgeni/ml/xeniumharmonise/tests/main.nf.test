
nextflow_process {

    name "Test Process ML_XENIUMHARMONISE"
    script "../main.nf"
    config "./nextflow.config"
    process "ML_XENIUMHARMONISE"

    tag "modules"
    tag "modules_sangercellgeni"
    tag "ml"
    tag "ml/xeniumharmonise"

    test("xeniumharmonise - dir") {

        
        when {
            process {
                """
                
                input[0] = [
                    'xeniumharmonise_test', // meta map
                    '/lustre/scratch127/cellgen/cellgeni/dn10/BK30-SKI-27-FO-3-S3/0043607 - BK30-SKI-27-FO-3-s3 - 2024-10-23 11.28.38.ndpi',
                    '/lustre/scratch127/cellgen/cellgeni/dn10/BK30-SKI-27-FO-3-S3/output-XETG00155__0043607__BK30-SKI-27-FO-3-S3__20241010__133127',
                    '/lustre/scratch127/cellgen/cellgeni/dn10/BK30-SKI-27-FO-3-S3/BK30-SKI-27-FO-3-S3_harmonised'
                ]
                """
            }
        }

        then {
            def harmonisedDir = process.out.harmonised_output?.getAt(0)?.getAt(1)
            assert harmonisedDir : "Expected harmonised directory output"

            // Define regex pattern: starts with a or d, followed by digits, ending in .tdb
            def pattern = ~/^[ad]\d+\.tdb$/

            // Collect matching files with full paths, sort them by path to ensure stable order
            def stableFiles = []
            file(harmonisedDir).eachFileRecurse { f ->
                if (f.name ==~ pattern) {
                    stableFiles << f
                }
            }
            stableFiles.sort { it.path }

            assertAll(
                { assert process.success },
                { assert snapshot(stableFiles, process.out.versions).match() }
                
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("xeniumharmonise - dir - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    'xeniumharmonise_test', // meta map
                    '/lustre/scratch127/cellgen/cellgeni/dn10/BK30-SKI-27-FO-3-S3/output-XETG00155__0043607__BK30-SKI-27-FO-3-S3__20241010__133127',
                    '/lustre/scratch127/cellgen/cellgeni/dn10/BK30-SKI-27-FO-3-S3/0043607 - BK30-SKI-27-FO-3-s3 - 2024-10-23 11.28.38.ndpi',
                    '/lustre/scratch127/cellgen/cellgeni/dn10/BK30-SKI-27-FO-3-S3/BK30-SKI-27-FO-3-S3_harmonised'
                ]
                """
            }
        }

        then {
            
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                
            )
        }

    }
}
