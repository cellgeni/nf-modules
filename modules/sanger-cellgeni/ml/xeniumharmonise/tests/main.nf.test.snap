{
    "xeniumharmonise - dir - stub": {
        "content": [
            {
                "0": [
                    [
                        "xeniumharmonise_test",
                        "xeniumharmonise_test_harmonised:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "harmonised_output": [
                    [
                        "xeniumharmonise_test",
                        "xeniumharmonise_test_harmonised:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-09-03T12:38:55.336376746"
    },
    "xeniumharmonise - dir": {
        "content": [
            [
                "a0.tdb:md5,5a4572165a283c0cc71e1c47ba4dc888",
                "d0.tdb:md5,3e7f0cde0abc12d2434153728f6a147c",
                "d1.tdb:md5,889d1506dfbbae363ec59869f9f06859",
                "a0.tdb:md5,66f3b18c04e9189e965d619b928213c7",
                "d0.tdb:md5,73aeaccfef7072718ff698d1c0e301a3",
                "d1.tdb:md5,c932b3b653bc1531b0081c30d8ddf964",
                "a0.tdb:md5,368cc4f14f63693cc380c7c36df169f2",
                "d0.tdb:md5,99524580c81016466eca9daab8fbe271",
                "d1.tdb:md5,6e76a1f32cee389571fce7e33867c56a",
                "a0.tdb:md5,a97dbd94a89ff19e9216b60a7c42653e",
                "d0.tdb:md5,92ad0291c67c0779d059e8c98a1c584e",
                "d1.tdb:md5,5fb5aa6a0904009c6fb34bfc58ee4c9f",
                "a0.tdb:md5,0259e2ad4a200c0cdc758f3102a42b2b",
                "d0.tdb:md5,385febb13e0c9ed50eb1a7fc5600b32f",
                "d1.tdb:md5,a7493472d2d324c865e7e9623b42d05c",
                "a0.tdb:md5,33da4da5a51498aa5c3ae781621371d2",
                "d0.tdb:md5,97162cf8b3e8299f8b99b70e8cd3dc5e",
                "a0.tdb:md5,8c05453f3c9548241467da6c4faad7ba",
                "a0.tdb:md5,8e932e8f706541f052df561d8532bbfd",
                "a1.tdb:md5,44fb177f909b1cbdcd4841da44cfc439",
                "a10.tdb:md5,3ab23119078792f30e4d3bab8772504b",
                "a2.tdb:md5,f3d044994b44cea27dcd02bbaa0364e8",
                "a3.tdb:md5,ce9bda82cba728d58f98fcf89bcbd092",
                "a4.tdb:md5,9f573d7482edb589355467a1a017f059",
                "a5.tdb:md5,89b749fe681fa2ee676f3f4f6f72c62a",
                "a6.tdb:md5,6407b22dbad0260e5544b2047af607cc",
                "a7.tdb:md5,d15df1113bc28bbf0cc65c8b321cd003",
                "a8.tdb:md5,a48b56d37fafb5be0cd772b770a513f5",
                "a9.tdb:md5,21c534fd775495ef82c36432eb5dde40",
                "d0.tdb:md5,7d9008175547ed973abb4523d27c64c3",
                "d1.tdb:md5,97162cf8b3e8299f8b99b70e8cd3dc5e",
                "d2.tdb:md5,6ef0d7c6e0e7a6109bd115703c49ba75",
                "d3.tdb:md5,f0f5f3261273a886a13a21a5160dcef9",
                "a0.tdb:md5,5e4d4be8606819ed9f213f84aaa14a6a",
                "a0.tdb:md5,93827b2ef8cc8bd20a1c67275a29af8e",
                "a0.tdb:md5,5f9fbfdc3faf007c4d002cc8ef21af4e",
                "a0.tdb:md5,344c7c22182be12dfe77ebe2e79f6877",
                "a0.tdb:md5,25fa4eb68e1d6c0df5c3760023c8a367",
                "a0.tdb:md5,a085dd67f39a799dfba0ae7c4bfb3ceb",
                "a0.tdb:md5,1579a85be92bb17e6c32f3a5dad5a9a6",
                "a0.tdb:md5,5174b0b5a4addf67544825136c58f62b",
                "a0.tdb:md5,b000b5f3092a2e1fae064856304e646b",
                "a0.tdb:md5,8b3bea5f085f303ee6ae70b855e836ea",
                "a0.tdb:md5,90374d10ee696ae792299fba487d0cc8",
                "a0.tdb:md5,2db3a67a51ee14d756f8448850653510",
                "a0.tdb:md5,1131fd17bedfefd7a43c527d05529c6b",
                "a0.tdb:md5,4b226246c549d34e0ca62b1d656b1c06",
                "a0.tdb:md5,edcbe14575c75155cca5410767961754",
                "a0.tdb:md5,998279301773062013a1da3a217360b4",
                "a0.tdb:md5,f2b905a5744df9cbb3cd543fc180f61e",
                "a0.tdb:md5,db83705573668c2e4b3b1926ece92f87",
                "a0.tdb:md5,0bc71a06fbeb071b936fadb3dda1fa61",
                "d0.tdb:md5,1eebc96e0a189b755a16163aba43508d",
                "a0.tdb:md5,fa15dbb212d16c4476278230a9320f28",
                "a1.tdb:md5,a43179da5e8305e8c763994dabb07b75",
                "d0.tdb:md5,9fa20506bd9bc0f8b12deeea0ea2cc6b",
                "d1.tdb:md5,df50700091b1db3d2c52e55e1816793e",
                "d2.tdb:md5,501dc697347b0ec5caf19ff60006a6c3",
                "d3.tdb:md5,2c5fe8824245c57e0b3d854a3152e038"
            ],
            [
                "versions.yml:md5,35f672d1fb01b542f667952c9dbb26fe"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-09-03T12:38:03.251329032"
    }
}